src/Simulation/EntryPointDriver/Azure/Azure.cs

```diff
@@ -271,26 +270,24 @@ private static void Log(AzureSettings settings, string message)
         /// </summary>
         /// <param name="settings">The Azure settings.</param>
         /// <returns>A Q# submitter.</returns>
-        /// <exception cref="ArgumentNullException"><paramref name="settings"/>.Target is null.</exception>
         private static IQSharpSubmitter? QSharpSubmitter(AzureSettings settings) => settings.Target switch
         {
-            null => throw new ArgumentNullException(nameof(settings), "Target is null."),
+            null => null,
             NoOpSubmitter.Target => new NoOpSubmitter(),
-            _ => null // TODO: Add a factory to Microsoft.Azure.Quantum.Client.
+            _ => SubmitterFactory.QSharpSubmitter("TODO", settings.Target, settings.CreateWorkspace(), settings.Storage)
         };
 
         /// <summary>
         /// Returns a QIR submitter for the target in the given Azure settings.
         /// </summary>
         /// <param name="settings">The Azure settings.</param>
         /// <returns>A QIR submitter.</returns>
-        /// <exception cref="ArgumentNullException"><paramref name="settings"/>.Target is null.</exception>
         private static IQirSubmitter? QirSubmitter(AzureSettings settings) => settings.Target switch
         {
-            null => throw new ArgumentNullException(nameof(settings), "Target is null"),
+            null => null,
             NoOpQirSubmitter.Target => new NoOpQirSubmitter(),
             NoOpSubmitter.Target => new NoOpSubmitter(),
-            _ => null // TODO: Add a factory to Microsoft.Azure.Quantum.Client.
+            _ => SubmitterFactory.QirSubmitter("TODO", settings.Target, settings.CreateWorkspace(), settings.Storage)
```

Comment:
@cesarzc, `AzureQirSubmitter` in qdk-private has a `providerId` parameter in its constructor, but I'm not sure what to put there. There isn't any provider option in the existing command-line options for Azure submission.

Reply:
This was based on our conversation where you said the providers would have the same constructor shape as `AzureQirSubmitter` - but it seems like the provider classes won't need their own provider ID, so will their constructors be:

```csharp
string target
IWorkspace workspace
string? storageConnectionString
```

instead?

Reply:
Sorry I missed that. Yes, there should be another class that uses `AzureQirSubmitter` but does not expose the `providerId` in its constructor. I'll create a `MicrosoftSubmitter` class following the same pattern that we have for IonQ and Honeywell. I'll open a PR shortly.